### PR TITLE
Added a bunch of mappings based of strings related to the custom dime…

### PR DESCRIPTION
**Pull request by Kroppeb**
Added a bunch of mappings related to the custom dimensions based on strings.

### CODE REVIEW

File: mappings/net/minecraft/world/dimension/StructuresConfig.mapping [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CLASS net/minecraft/class_5311 net/minecraft/world/dimension/StructuresConfig
+	FIELD field_24821 structuresConfigCodec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/StructuresConfig.mapping]:
```suggestion
	FIELD field_24821 CODEC Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/dimension/StructuresConfig.mapping [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	FIELD field_24822 structureConfig Lcom/google/common/collect/ImmutableMap;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/StructuresConfig.mapping]:
```suggestion
	FIELD field_24822 DEFAULT_STRUCTURES Lcom/google/common/collect/ImmutableMap;
```

File: mappings/net/minecraft/world/dimension/StructuresConfig.mapping [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	FIELD field_24823 strongHoldConfig Lnet/minecraft/class_5313;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/StructuresConfig.mapping]:
```suggestion
	FIELD field_24823 STRONGHOLD_CONFIG Lnet/minecraft/class_5313;
```

File: mappings/net/minecraft/world/dimension/StructureConfig.mapping [hunk 4]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_5314 net/minecraft/world/dimension/StructureConfig
+	FIELD field_24917 structureConfigCodec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/StructureConfig.mapping]:
```suggestion
	FIELD field_24917 CODEC Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/dimension/StrongholdConfig.mapping [hunk 5]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_5313 net/minecraft/world/dimension/StrongholdConfig
+	FIELD field_24913 strongholdConfigCodec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/StrongholdConfig.mapping]:
```suggestion
	FIELD field_24913 CODEC Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/dimension/SlideConfig.mapping [hunk 6]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_5310 net/minecraft/world/dimension/SlideConfig
+	FIELD field_24817 slideConfigCodec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/SlideConfig.mapping]:
```suggestion
	FIELD field_24817 CODEC Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping [hunk 7]
```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/class_5308 net/minecraft/world/dimension/NoiseSamplingConfig
+	FIELD field_24799 noiseSamplingConfig Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping]:
```suggestion
	FIELD field_24799 CODEC Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/dimension/NoiseConfig.mapping [hunk 8]
```diff
@@ -0,0 +1,39 @@
+CLASS net/minecraft/class_5309 net/minecraft/world/dimension/NoiseConfig
+	FIELD field_24804 noiseConfigCodec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseConfig.mapping]:
```suggestion
	FIELD field_24804 CODEC Lcom/mojang/serialization/Codec;
```

**Review by liach — CHANGES_REQUESTED** [review]
Fyi we use upper snake case for static final fields disregarding whether these fields are mutable or not.

File: mappings/net/minecraft/world/dimension/NoiseConfig.mapping [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	METHOD method_28584 getTopSlide ()Lnet/minecraft/class_5310;
+	METHOD method_28585 getBottomSlide ()Lnet/minecraft/class_5310;
+	METHOD method_28586 getSizeHorizontal ()I
+	METHOD method_28587 getSizeVertical ()I
+	METHOD method_28588 getDensityFactor ()D
+	METHOD method_28589 getDensityOffset ()D
+	METHOD method_28590 isSimplexSurfaceNoise ()Z
+	METHOD method_28591 isRandomDensityOffset ()Z
+	METHOD method_28592 isIslandNoiseOverride ()Z
+	METHOD method_28593 getAmplified ()Z
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseConfig.mapping]:
```suggestion
	METHOD method_28593 isAmplified ()Z
```

File: mappings/net/minecraft/world/dimension/NoiseConfig.mapping [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	METHOD method_28583 getSampling ()Lnet/minecraft/class_5308;
+	METHOD method_28584 getTopSlide ()Lnet/minecraft/class_5310;
+	METHOD method_28585 getBottomSlide ()Lnet/minecraft/class_5310;
+	METHOD method_28586 getSizeHorizontal ()I
+	METHOD method_28587 getSizeVertical ()I
+	METHOD method_28588 getDensityFactor ()D
+	METHOD method_28589 getDensityOffset ()D
+	METHOD method_28590 isSimplexSurfaceNoise ()Z
+	METHOD method_28591 isRandomDensityOffset ()Z
+	METHOD method_28592 isIslandNoiseOverride ()Z
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseConfig.mapping]:
```suggestion
	METHOD method_28592 hasIslandNoiseOverride ()Z
```

File: mappings/net/minecraft/world/dimension/NoiseConfig.mapping [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+		ARG 14 amplified
+	METHOD method_28581 getHeight ()I
+	METHOD method_28583 getSampling ()Lnet/minecraft/class_5308;
+	METHOD method_28584 getTopSlide ()Lnet/minecraft/class_5310;
+	METHOD method_28585 getBottomSlide ()Lnet/minecraft/class_5310;
+	METHOD method_28586 getSizeHorizontal ()I
+	METHOD method_28587 getSizeVertical ()I
+	METHOD method_28588 getDensityFactor ()D
+	METHOD method_28589 getDensityOffset ()D
+	METHOD method_28590 isSimplexSurfaceNoise ()Z
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseConfig.mapping]:
```suggestion
	METHOD method_28590 hasSimplexSurfaceNoise ()Z
```

File: mappings/net/minecraft/world/dimension/NoiseConfig.mapping [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+	METHOD method_28591 isRandomDensityOffset ()Z
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseConfig.mapping]:
```suggestion
	METHOD method_28591 hasRandomDensityOffset ()Z
```

File: mappings/net/minecraft/world/dimension/NoiseConfig.mapping [hunk 13]
```diff
@@ -0,0 +1,39 @@
+CLASS net/minecraft/class_5309 net/minecraft/world/dimension/NoiseConfig
+	FIELD field_24804 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24805 height I
+	FIELD field_24806 sampling Lnet/minecraft/class_5308;
+	FIELD field_24807 topSlide Lnet/minecraft/class_5310;
+	FIELD field_24808 bottomSlide Lnet/minecraft/class_5310;
+	FIELD field_24809 sizeHorizontal I
+	FIELD field_24810 sizeVertical I
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseConfig.mapping]:
```suggestion
	FIELD field_24809 horizontalSize I
	FIELD field_24810 verticalSize I
```

Don't just follow mojang's trashy name suite

File: mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping [hunk 14]
```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/class_5308 net/minecraft/world/dimension/NoiseSamplingConfig
+	FIELD field_24799 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24800 xzScale D
+	FIELD field_24801 yScale D
+	FIELD field_24802 xzFactor D
+	FIELD field_24803 yFactor D
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping]: imo horizontal/vertical factor/scale would be better

File: mappings/net/minecraft/world/dimension/StructuresConfig.mapping [hunk 15]
```diff
@@ -0,0 +1,11 @@
+CLASS net/minecraft/class_5311 net/minecraft/world/dimension/StructuresConfig
+	FIELD field_24821 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24822 DEFAULT_STRUCTURES Lcom/google/common/collect/ImmutableMap;
+	FIELD field_24823 DEFAULT_STRONGHOLD_CONFIG Lnet/minecraft/class_5313;
```
Comment (liach) [inline mappings/net/minecraft/world/dimension/StructuresConfig.mapping]:
```suggestion
	FIELD field_24823 DEFAULT_STRONGHOLD Lnet/minecraft/class_5313;
```

File: mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping (hunk 14 above)
Comment (misode) [reply mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping]: They're called like this in the config files. https://gist.github.com/RainWarrior/2c32848aad78910a3ca148b82c91f7c2#file-worldgen_settings-json-L204

Comment (Kroppeb) [reply mappings/net/minecraft/world/dimension/NoiseSamplingConfig.mapping]: Yeah, I'm just basing this off the codecs.

File: mappings/net/minecraft/world/dimension/StructuresConfig.mapping (hunk 15 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/dimension/StructuresConfig.mapping]: I prefer the `_CONFIG` at the end, as this isn't a stronghold but a stronghold config

Comment (liach) [reply mappings/net/minecraft/world/dimension/StructuresConfig.mapping]: @Kroppeb no, this class is already a config holder, and the actual instance field that holds the stronghold config you've already named `stronghold` below. You need consistency here. Should you add `config` suffix for this field, you should do so for the other 3 fields ((default)structure/stronghold) and apparently `config` suffix is unnecessary as this class is already `StructuresConfig`

**Review by liach — APPROVED** [review]

**Comment by YanisBft** [comment]
These classes are in the wrong package

**Review by liach — CHANGES_REQUESTED** [review]
yanis got a great point, these classes should not be in world.dimension package at all

**Comment by Kroppeb** [comment]
In which package should they be? I added these mappings there after a suggestion in the discord, and it seemed the most logical to place them together with `DimentionType`

**Comment by liach** [comment]
To `net/minecraft/world/gen/chunk` package. These have **nothing** to do with dimensions.

**Review by liach — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Comment by liach** [comment]
If this is to be merged the existing class_5309.mapping etc. need to be removed (they exist for the codec field)

**Review by Shnupbups — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Review by liach — CHANGES_REQUESTED** [review]
Mind remove `class_5311.mapping` as well

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**